commons/src/main/kotlin/com/simplemobiletools/commons/views/bottomactionmenu/BottomActionMenuItemPopup.kt

```diff
@@ -0,0 +1,247 @@
+            otherHeights += padding + listPadding
+        }
+
+        return listContent + otherHeights
+    }
+
+    private fun updateContentWidth(width: Int) {
+        val popupBackground = popup.background
+        dropDownWidth = if (popupBackground != null) {
+            popupBackground.getPadding(tempRect)
+            tempRect.left + tempRect.right + width
+        } else {
+            width
+        }
+    }
+
+    /**
+     * @see androidx.appcompat.widget.DropDownListView.measureHeightOfChildrenCompat
+     */
+    private fun measureHeightOfChildrenCompat(maxHeight: Int): Int {
+
+        val parent = FrameLayout(context)
+        val widthMeasureSpec = MeasureSpec.makeMeasureSpec(dropDownWidth, MeasureSpec.EXACTLY)
+
+        // Include the padding of the list
+        var returnedHeight = 0
+
+        val count = popupListAdapter.count
+
+        var child: View? = null
+        var viewType = 0
+        for (i in 0 until count) {
+            val positionType = popupListAdapter.getItemViewType(i)
+            if (positionType != viewType) {
+                child = null
+                viewType = positionType
+            }
+            child = popupListAdapter.getView(i, child, parent)
+
+            // Compute child height spec
+            val heightMeasureSpec: Int
+            var childLp: ViewGroup.LayoutParams? = child.layoutParams
+
+            if (childLp == null) {
+                childLp = ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT)
+                child.layoutParams = childLp
+            }
+
+            heightMeasureSpec = if (childLp.height > 0) {
+                MeasureSpec.makeMeasureSpec(
+                    childLp.height,
+                    MeasureSpec.EXACTLY
+                )
+            } else {
+                MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED)
+            }
+            child.measure(widthMeasureSpec, heightMeasureSpec)
+            // Since this view was measured directly against the parent measure
+            // spec, we must measure it again before reuse.
+            child.forceLayout()
+
+            val marginLayoutParams = childLp as? ViewGroup.MarginLayoutParams
+            val topMargin = marginLayoutParams?.topMargin ?: 0
+            val bottomMargin = marginLayoutParams?.bottomMargin ?: 0
+            val verticalMargin = topMargin + bottomMargin
+
+            returnedHeight += child.measuredHeight + verticalMargin
+
+            if (returnedHeight >= maxHeight) {
+                // We went over, figure out which height to return.  If returnedHeight >
+                // maxHeight, then the i'th position did not fit completely.
+                return maxHeight
+            }
+        }
+
+        // At this point, we went through the range of children, and they each
+        // completely fit, so return the returnedHeight
+        return returnedHeight
+    }
+
+
```

Comment:
remove such double empty lines please and dont forget formatting the files with Ctrl + Shift + L